streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableFilter.java

```diff
@@ -77,6 +77,9 @@ public void process(K key, Change<V> change) {
             V newValue = computeValue(key, change.newValue);
             V oldValue = sendOldValues ? computeValue(key, change.oldValue) : null;
 
+            if (sendOldValues && oldValue == null && newValue == null)
```

Comment:
Just to double-check because the JIRA ticket is a bit difficult to follow: Is this single condition sufficient to cover the following discussion/flow in the JIRA ticket?

> 3) If "send old value" is enabled, then there are a couple of cases we can consider:
>
> a. If old value is <key: null> and new value is <key: not-null>, and the filter predicate return false for the new value, then in this case it is safe to optimize and not returning anything to the downstream operator, since in this case we know there is no value for the key previously anyways; otherwise we send the original pair.
>
> b. If old value is <key: not-null> and new value is <key: null>, indicating to delete this key, and the filter predicate return false for the old value, then in this case it is safe to optimize and not returning anything to the downstream operator, since we know that the old value has already been filtered in a previous message; otherwise we send the original pair.
>
> c. If both old and new values are not null, and:
>   1) predicate return true on both, send the original pair;
>   2) predicate return false on both, we can optimize and do not send anything;
>   3) predicate return true on old and false on new, send the key: {old -> null};
>   4) predicate return false on old and true on new, send the key: {null -> new};